network/framework/src/peer.rs

```diff
@@ -0,0 +1,861 @@
+// Copyright © Aptos Foundation
+
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::runtime::Handle;
+use tokio::sync::mpsc::Receiver;
+use crate::protocols::wire::messaging::v1::{ErrorCode, MultiplexMessage, MultiplexMessageSink, MultiplexMessageStream, NetworkMessage, STREAM_FRAGMENT_OVERHEAD_BYTES, STREAM_HEADER_OVERHEAD_BYTES};
+use futures::io::{AsyncRead,AsyncReadExt,AsyncWrite};
+use futures::StreamExt;
+use futures::SinkExt;
+use futures::stream::Fuse;
+use tokio::sync::mpsc::error::TryRecvError;
+use aptos_config::config::{NetworkConfig, RoleType};
+use aptos_config::network_id::{NetworkContext, NetworkId, PeerNetworkId};
+use aptos_logger::{error, info, warn};
+use aptos_metrics_core::{IntCounter, IntCounterVec, register_int_counter_vec};
+use crate::application::ApplicationCollector;
+use crate::application::interface::{OpenRpcRequestState, OutboundRpcMatcher};
+use crate::application::storage::PeersAndMetadata;
+use crate::ProtocolId;
+use crate::protocols::network::{Closer, OutboundPeerConnections, PeerStub, ReceivedMessage};
+use crate::protocols::stream::{StreamFragment, StreamHeader, StreamMessage};
+use crate::transport::ConnectionMetadata;
+use once_cell::sync::Lazy;
+#[cfg(test)]
```

Comment:
nit: if you define a test module below, it should make all of these imports cleaner.